Source/Core/MacUpdater/MacUI.mm

```diff
@@ -138,20 +139,30 @@ void run_on_main(std::function<void()> fnc)
 {
 }
 
-Platform::BuildInfo::BuildInfo(const std::string& content)
+bool Platform::VersionCheck(const std::vector<TodoList::UpdateOp>& to_update,
+                            const std::string& install_base_path, const std::string& temp_dir,
+                            FILE* log_fp)
 {
-  map = {{"OSMinimumVersionMacOS", ""}};
-  Parse(content);
-}
+  std::string path = temp_dir + "/Dolphin.app";
 
-bool Platform::VersionCheck(const BuildInfo& this_build_info, const BuildInfo& next_build_info)
-{
-  // TODO implement OS Minimum Version check
-  // It should go something like this:
-  // auto target_version = next_build_info.GetVersion("OSMinimumVersionMacOS");
-  // if (!target_version.has_value() || current_version >= target_version)
-  //   return true;
-  // show error
-  // return false;
+  NSURL* bundle_url = [NSURL fileURLWithPath:[NSString stringWithCString:path.c_str()]];
+  CFDictionaryRef bundle_info_dict = CFBundleCopyInfoDictionaryInDirectory((CFURLRef)bundle_url);
+  if (!bundle_info_dict)
+    return true;
+  auto min_version_str =
+      (CFStringRef)CFDictionaryGetValue(bundle_info_dict, CFSTR("LSMinimumSystemVersion"));
+  if (!min_version_str)
+    return true;
+
+  NSArray* components = [min_version_str componentsSeparatedByString:@"."];
+  NSOperatingSystemVersion next_version{[[components objectAtIndex:0] integerValue],
```

Comment:
You could use some syntactic sugar here (optional): `components[0]`, `components[1]`, `components[2]`